.ci/publish-images.sh

```diff
@@ -0,0 +1,267 @@
+#!/bin/bash -eu
+
+# Publish any versions of the docker image not yet pushed to jenkins/jenkins
+# Arguments:
+#   -n dry run, do not publish images
+#   -d debug
+#   -f force, will publish images no matter what
+
+set -eou pipefail
+
+. jenkins-support
+
+: "${DOCKERHUB_ORGANISATION:=jenkins4eval}"
+: "${DOCKERHUB_REPO:=jenkins}"
+
+JENKINS_REPO="${DOCKERHUB_ORGANISATION}/${DOCKERHUB_REPO}"
+
+cat <<EOF
+Docker repository in Use:
+* JENKINS_REPO: ${JENKINS_REPO}
+EOF
+
+#This is precautionary step to avoid accidental push to official jenkins image
+if [[ "$DOCKERHUB_ORGANISATION" == "jenkins" ]]; then
+    echo "Experimental docker image should not published to jenkins organization , hence exiting with failure";
+    exit 1;
+fi
+
+docker-login() {
+    docker login --username ${DOCKERHUB_USERNAME} --password ${DOCKERHUB_PASSWORD}
```

Comment:
```suggestion
    docker login
```

We provide a DOCKER_CONFIG env variable in the Jenkinsfile setup, so the username and password should not be needed.

Reply:
Maybe we should document that as a comment in this function.

Reply:
I added a comment in my new branch:

```
    # Making use of the credentials store in `config.json`

```